### PR TITLE
better fallback URL, allow symlinks, fix unlink with protect, rebalance endpoint

### PR DESCRIPTION
**Pull request by gregjhogan**

### CODE REVIEW

File: src/server.go [hunk 1]
```diff
@@ -101,25 +101,24 @@ func (a *App) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
       if a.fallback == "" {
         w.Header().Set("Content-Length", "0")
         w.WriteHeader(404)
         return
-      } else {
-        // fall through to fallback
-        volume = a.fallback
       }
+      // fall through to fallback
+      remote = fmt.Sprintf("http://%s%s", a.fallback, key)
```
Comment (geohot) [inline src/server.go]: I don't like this. What if key has non URL chars in it?

File: src/server.go [hunk 2]
```diff
@@ -184,7 +183,7 @@ func (a *App) ServeHTTP(w http.ResponseWriter, r *http.Request) {
       return
     }
 
-    if a.protect && rec.deleted == NO {
+    if !unlink && a.protect && rec.deleted == NO {
```
Comment (geohot) [inline src/server.go]: Wow, did UNLINK just not work? We should add a test.

File: src/server.go (hunk 1 above)
Comment (gregjhogan) [reply src/server.go]: I think whatever was sent for the URL in the original GET request should be fine to pass through to the fallback endpoint.  I think base64 encoding the name only makes sense when writing a file out to disk (don't want slashes in file name).  Also, `key2path` includes the folder structure, which the fallback shouldn't have to deal with.

File: src/server.go (hunk 2 above)
Comment (gregjhogan) [reply src/server.go]: yeah tests are MIA for unlink and protect

Comment (Georgantas) [reply src/server.go]: +1

**Pull request merged**